spec/services/analytics_spec.rb

```diff
@@ -42,66 +59,31 @@
         'my branch',
       )
 
-      analytics_hash = {
-        event_properties: {},
-        user_id: current_user.uuid,
-        locale: I18n.locale,
-        git_sha: IdentityConfig::GIT_SHA,
-        git_branch: IdentityConfig::GIT_BRANCH,
-        new_session_path: true,
-        new_session_success_state: true,
-        success_state: success_state,
-        new_event: true,
-        path: path,
-        session_duration: nil,
-      }
-
-      expect(ahoy).to receive(:track).
-        with('Trackable Event', analytics_hash.merge(request_attributes))
+      expect(ahoy).to receive(:track).with('Trackable Event', expected_data_to_analytics)
 
       analytics.track_event('Trackable Event')
     end
 
     it 'does not track unique events and paths when an event fails' do
-      analytics_hash = {
-        event_properties: { success: false },
-        user_id: current_user.uuid,
-        locale: I18n.locale,
-        git_sha: IdentityConfig::GIT_SHA,
-        git_branch: IdentityConfig::GIT_BRANCH,
-        new_session_path: nil,
-        new_session_success_state: nil,
-        success_state: success_state,
-        new_event: nil,
-        path: path,
-        session_duration: nil,
-      }
-
-      expect(ahoy).to receive(:track).
-        with('Trackable Event', analytics_hash.merge(request_attributes))
+      expected_data_to_analytics.merge!(
+        {
+          new_event: nil,
+          new_session_path: nil,
+          new_session_success_state: nil,
+          event_properties: { success: false },
+        },
+      )
+
+      expect(ahoy).to receive(:track).with('Trackable Event', expected_data_to_analytics)
```

Comment:
it's a "six of one, half a dozen of the other" kind of situation, but I would not bother with mutating the shared hash here (it doesn't matter because a `let` block is reset between spec runs) but I think it might be clearer to do an inline merge:

(and we can drop the optional curly braces)

```suggestion
      expect(ahoy).to receive(:track).with(
        'Trackable Event',
        expected_data_to_analytics.merge(
          new_event: nil,
          new_session_path: nil,
          new_session_success_state: nil,
          event_properties: { success: false },
      ),
    )
```